miflora-mqtt-daemon.py

```diff
@@ -307,6 +307,15 @@ def flores_to_openhab_items(flores, reporting_mode):
             payload['name'] = "{} {}".format(flora_name, sensor.title())
             if 'device_class' in params:
                 payload['device_class'] = params['device_class']
+            payload['unique_id'] = "{}-{}".format(flora['mac'].lower().replace(":", ""), sensor)
+            payload['device'] = {
+                    'identifiers' : ["MiFlora{}".format(flora['mac'].lower().replace(":", ""))],
```

Comment:
Out of curiosity, what is this used for?

Reply:
It is necessary info for HA Device registry so Miflora Device is created upon HA MQTT discovery with its device connection (BTLE MAC address) and then all sensors like temperature, lux, moisture etc. are linked with this new device and identified with by theirs unique_id (btle_mac-temperature, btle_mac-light, ....). Details: https://www.home-assistant.io/integrations/sensor.mqtt/. There is also ha-core github discussion with mqtt json example: https://github.com/home-assistant/core/pull/16943#issue-219014397.

Reply:
Sorry. Of course I'm aware of the device discovery feature :) I was asking about the `identifiers` specifically. Documentation says

> A list of IDs that uniquely identify the device. For example a serial number.

But would you by any chance know what it is actually used for? I wonder whether or not more than one element per sensor would make sense... just curious :)

Reply:
If we are talking about unique_id: it is necessary to be unique for each entity(=sensor) - probably some unique index in device registry db/table. Each entity with unique_id must be then linked by 'device' section with its physical device (control unit with multiple sensors). If you look into new miflora the MQTT topic, then each sensor (temperature, lux, moisture, conductivity, battery) has different unique_id, but same parent device. That is how relationship is defined and HA know the connections.

Details:
https://developers.home-assistant.io/docs/device_registry_index/

Reply:
regarding multiple Identifiers - don't know why there should be more than one unique identifiers... can't think of some use case when you would need more identifiers. But I'm not a HA guru... :)

Reply:
Nope me neither. Maybe something for a later point. Feels like this whole concept of devices becomes more and more important in HA